main.go

```diff
@@ -1,28 +1,184 @@
 package main
 
 import (
-	"fmt"
+	"github.com/juju/errgo"
 
 	"github.com/xh3b4sd/anna/core"
 	"github.com/xh3b4sd/anna/gateway"
+	"github.com/xh3b4sd/anna/impulse"
+	"github.com/xh3b4sd/anna/log"
+	"github.com/xh3b4sd/anna/network"
+	"github.com/xh3b4sd/anna/neuron"
 	"github.com/xh3b4sd/anna/server"
 )
 
+var (
+	maskAny = errgo.MaskFunc(errgo.Any)
+)
+
 func main() {
-	fmt.Printf("%s\n", "hello, I am Anna")
+	//
+	// deps
+	//
+	newLog := log.NewLog(log.DefaultConfig())
+	newLog.V(7).Infof("%s", "hello, I am Anna")
+
+	newTextGateway := gateway.NewGateway()
+
+	//
+	// core
+	//
+	newImpulseConfig := impulse.DefaultConfig()
+	newImpulseConfig.Log = newLog
+	newImpulse := impulse.NewImpulse(newImpulseConfig)
+
+	newFirstConfig := neuron.DefaultFirstNeuronConfig()
+	newFirstConfig.Log = newLog
+	newFirst := neuron.NewFirstNeuron(newFirstConfig)
+
+	newJobConfig := neuron.DefaultJobNeuronConfig()
+	newJobConfig.Log = newLog
+	newJob := neuron.NewJobNeuron(newJobConfig)
 
-	textGateway := gateway.NewGateway()
+	newCharacterConfig := neuron.DefaultCharacterNeuronConfig()
+	newCharacterConfig.Log = newLog
+	newCharacter := neuron.NewCharacterNeuron(newCharacterConfig)
+
+	newNetworkConfig := network.DefaultConfig()
+	newNetworkConfig.Log = newLog
+	newNetwork := network.NewNetwork(newNetworkConfig)
 
-	fmt.Printf("%s\n", "booting core")
 	newCoreConfig := core.DefaultConfig()
-	newCoreConfig.TextGateway = textGateway
+	newCoreConfig.Log = newLog
+	newCoreConfig.TextGateway = newTextGateway
 	newCore := core.NewCore(newCoreConfig)
+
+	impulseState, err := newImpulse.GetState("default")
+	if err != nil {
+		newLog.V(1).Errorf("%#v", maskAny(err))
+	}
+	impulseState = impulseState.Copy()
+	impulseState.SetImpulse(newImpulse)
+	impulseState.SetBytes("impulse-id", []byte(newImpulse.GetObjectID()))
+	impulseState.SetNeuron(newFirst)
+	impulseState.SetBytes("first-id", []byte(newFirst.GetObjectID()))
+	impulseState.SetNeuron(newJob)
+	impulseState.SetBytes("job-id", []byte(newJob.GetObjectID()))
+	impulseState.SetNeuron(newCharacter)
+	impulseState.SetBytes("character-id", []byte(newCharacter.GetObjectID()))
+	impulseState.SetNetwork(newNetwork)
+	impulseState.SetBytes("network-id", []byte(newNetwork.GetObjectID()))
+	impulseState.SetCore(newCore)
+	impulseState.SetBytes("core-id", []byte(newCore.GetObjectID()))
+	newImpulse.SetState("init", impulseState)
+
+	firstState, err := newFirst.GetState("default")
+	if err != nil {
+		newLog.V(1).Errorf("%#v", maskAny(err))
+	}
+	firstState = firstState.Copy()
```

Comment:
It would be cool to improve the design of different neuron types. This initialization is horrible. There is also a lot of duplicated code to fulfil the interfaces. Maybe some behavior implementation would solve the problem. Need to think about it.